galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerManagerImpl.java

```diff
@@ -170,6 +174,26 @@ public IDockerEngine generateDockerEngine(Field field, List<Annotation> annotati
         return this.getDockerEngine(annotationServer.dockerEngineTag());
     }
 
+    @GenerateAnnotatedField(annotation = DockerContainerConfig.class)
+    public IDockerContainerConfig generateDockerContainerConfig(Field field, List<Annotation> annotations) throws DockerManagerException {
+        List<IDockerVolume> volumes = new ArrayList<>();
+        DockerContainerConfig config = field.getAnnotation(DockerContainerConfig.class);
+
+        for (DockerVolume volumeAnnotation : config.dockerVolumes()) {
+            volumes.add(generateDockerVolumme(volumeAnnotation));
+        }
+        return new DockerContainerConfigImpl(volumes);
+    }
+
+    @GenerateAnnotatedField(annotation =  DockerVolume.class)
+    public IDockerVolume generateDockerVolumme(DockerVolume annotation) throws DockerManagerException {
```

Comment:
I cannot see any clean up of the volumes at the end of the test

Reply:
They are discarded (if tagged too) when the dockerenviroment is discarded as part of provision stop